src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,42 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDaoImpl storage;
+
+    public RegistrationServiceImpl(StorageDaoImpl storage) {
+        this.storage = storage;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new RegistrationException("Login and password cannot be null");
```

Comment:
Don't create one big if with all conditions. Better to split one big if into several small ones. This way, the exception messages can be more informative and specific to the condition that failed. For instance, separate the checks for 'login' and 'password' into different if statements.